File: backend/components/metiers/services_personne/tests/test_api.py
from fastapi.testclient import TestClient
from ..views import router
from fastapi import FastAPI

app = FastAPI()
app.include_router(router)

client = TestClient(app)

def test_list_environnements():
    response = client.get("/environnements")
    assert response.status_code == 200
    assert "environnements" in response.json()
    assert "total" in response.json()

def test_create_environnement():
    data = {"nom": "Test Env", "description": "desc", "type": "zone"}
    response = client.post("/environnements", json=data)
    assert response.status_code == 201
    assert response.json()["nom"] == "Test Env"

File: backend/components/metiers/science/tests/test_api.js
// test_api.js – Tests avancés de l'API Environnement (Node.js/Jest)
const request = require('supertest');
const app = require('../../index'); // Adapter selon l'entrée réelle de l'app

describe('API Environnement', () => {
  it('GET /environnements doit retourner la liste', async () => {
    const res = await request(app).get('/environnements');
    expect(res.statusCode).toBe(200);
    expect(res.body).toHaveProperty('environnements');
    expect(Array.isArray(res.body.environnements)).toBe(true);
  });

  it('POST /environnements crée une entité', async () => {
    const data = { nom: 'Test JS', description: 'desc', type: 'zone' };
    const res = await request(app).post('/environnements').send(data);
    expect(res.statusCode).toBe(201);
    expect(res.body.nom).toBe('Test JS');
  });
});
